doc/guide/sample_ml_app.hpp

```diff
@@ -0,0 +1,194 @@
+- Under Build Events > Post-Build Event > Command Line add:
+@code
+ - xcopy /y "C:\mlpack\mlpack-3.0.2\build\Debug\mlpack.dll" $(OutDir)
+ - xcopy /y "C:\mlpack\mlpack-3.0.2\packages\OpenBLAS.0.2.14.1\lib\native\bin\x64\*.dll" $(OutDir)
+@endcode
+
+@note Recent versions of Visual Studio set "Conformance Mode" enabled by default. This causes some issues with
+the armadillo library. If you encounter this issue, disable "Conformance Mode" under C/C++ > Language.
+
+@section sample_app_goal The app goal
+
+This app aims to exercise an end-to-end machine learning workflow. We will cover:
+
+- Loading and preparing a dataset
+- Training (using Random Forest as example)
+- Computing the training accuracy
+- Cross-Validation using K-Fold
+- Metrics gathering (accuracy, precision, recall, F1)
+- Saving the trained model to disk
+- Loading the model
+- Classifying a new sample
+
+@section sample_headers_namespaces Headers and namespaces
+
+For this app, we will need to include the following headers (i.e. add into stdafx.h):
+
+@code
+#include "mlpack/core.hpp"
+#include "mlpack/methods/random_forest/random_forest.hpp"
+#include "mlpack/methods/decision_tree/random_dimension_select.hpp"
+#include "mlpack/core/cv/k_fold_cv.hpp"
+#include "mlpack/core/cv/metrics/accuracy.hpp"
+#include "mlpack/core/cv/metrics/precision.hpp"
+#include "mlpack/core/cv/metrics/recall.hpp"
+#include "mlpack/core/cv/metrics/F1.hpp"
+@endcode
+
+Also, we will use the following namespaces:
+
+@code
+using namespace arma;
+using namespace mlpack;
+using namespace mlpack::tree;
+using namespace mlpack::cv;
+@endcode
+
+@section sample_load_dataset Loading the dataset
+
+First step is about loading the dataset. Here we load a CSV dataset 
+(i.e.: mlpack/tests/data/german.csv), assuming the labels don't require normalization.
+
+@code
+bool loaded = mlpack::data::Load("data/german.csv", dataset);
+if (!loaded)
+	return -1;
+@endcode
+
+Then we need to extract the labels from the last dimension of the dataset and remove the
+labels from the training set:
+
+@code
+Row<size_t> labels;
+labels = conv_to<Row<size_t>>::from(dataset.row(dataset.n_rows - 1));
+dataset.shed_row(dataset.n_rows - 1);
+@endcode
+
+We now have our dataset ready for training.
+
+@section sample_training Training
+
+This app will use a Random Forest classifier. At first we define the classifier parameters and then
+we create the classifier to train it.
+
+@code
+const size_t numClasses = 2;
+const size_t minimumLeafSize = 5;
+const size_t numTrees = 10;
+
+RandomForest<GiniGain, RandomDimensionSelect> rf;
+
+rf = RandomForest<GiniGain, RandomDimensionSelect>(dataset, labels,
+	numClasses, numTrees, minimumLeafSize);
+@endcode
+
+Now that the training is completed, we quickly compute the training accuracy:
+
+@code
+Row<size_t> predictions;
+rf.Classify(dataset, predictions);
+const size_t correct = arma::accu(predictions == labels);
+printf("\nTraining Accuracy: %f", (double(correct) / double(labels.n_elem)));
```

Comment:
hm, should we use std::cout here? Usually we don't use ``printf`` in the codebase.

Reply:
Changed all the printf to cout - thanks